### PR TITLE
Parser cleanup (remove dependency on Solidity AST)

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

**Comment by axic** [comment]
This should be merged after #2224.

File: libsolidity/parsing/Parser.cpp [hunk 1]
```diff
@@ -1437,5 +1437,61 @@ ASTPointer<ParameterList> Parser::createEmptyParameterList()
 	return nodeFactory.createNode<ParameterList>(vector<ASTPointer<VariableDeclaration>>());
 }
 
+Token::Value Parser::expectAssignmentOperator()
+{
+	Token::Value op = m_scanner->currentToken();
+	if (!Token::isAssignmentOp(op))
+	{
+		if (Token::isElementaryTypeName(op)) //for the sake of accuracy in reporting
```
Comment (chriseth) [inline libsolidity/parsing/Parser.cpp]: Can you pull out the common parts in the clauses here?

File: libsolidity/parsing/Parser.cpp [hunk 2]
```diff
@@ -1437,5 +1437,61 @@ ASTPointer<ParameterList> Parser::createEmptyParameterList()
...
+		{
+			ElementaryTypeNameToken elemTypeName = m_scanner->currentElementaryTypeNameToken();
+			fatalParserError(
+				string("Expected identifier, got '") +
+				elemTypeName.toString() +
+				string("'")
+			);
+		}
+		else
+			fatalParserError(
```
Comment (chriseth) [inline libsolidity/parsing/Parser.cpp]: Again, please combine the common parts.

**Review by chriseth — CHANGES_REQUESTED** [review]

**Comment by axic** [comment]
@chriseth updated

**Review by chriseth — APPROVED** [review]

**Pull request merged**